### PR TITLE
Move refill_unblock_check lemmas

### PR DESCRIPTION
**Pull request by mktnk3**
This PR moves `refill_unblock_check` lemmas in the way needed by the rest of the “add refill_unblock_check” changes.

I would like to this merged now so that I don’t need to keep redoing this... 😅 

Contains a bit of cleanup of unused lemmas I noticed along the way.

### CODE REVIEW

File: proof/invariant-abstract/DetSchedSchedule_AI.thy [hunk 1]
```diff
@@ -3628,6 +3628,798 @@ lemma set_thread_state_valid_sched:
...
+   apply (wpsimp wp: set_refills_wp get_refills_wp)
+   apply (clarsimp simp: vs_all_heap_simps refills_unat_sum_def)
+  apply (wpsimp wp: refill_head_overlapping_loop_hd_r_amount)
+  done
+
+lemma refill_unblock_check_budget_sufficient[wp]:
+  "\<lbrace>budget_sufficient tcb_ptr and active_sc_valid_refills
+    and (\<lambda>s. \<forall>scp. bound_sc_tcb_at ((=) (Some scp)) tcb_ptr s \<longrightarrow> valid_refills scp s)\<rbrace>
+   refill_unblock_check sc_ptr
+   \<lbrace>\<lambda>xc s. budget_sufficient tcb_ptr s\<rbrace>"
```
Comment (michaelmcinerney) [inline proof/invariant-abstract/DetSchedSchedule_AI.thy:3822]: Maybe nicer to ignore the return value here

File: proof/invariant-abstract/DetSchedSchedule_AI.thy [hunk 2]
```diff
@@ -3628,6 +3628,798 @@ lemma set_thread_state_valid_sched:
...
+  apply (intro conjI impI)
+   apply (simp add: list.set_sel(2))
+  apply (clarsimp simp: refills_unat_sum_def)
+  apply (case_tac "sc_refills sc"; clarsimp)
+  done
+
+lemma refill_unblock_check_active_sc_valid_refills:
+  "\<lbrace>active_sc_valid_refills
+    and current_time_bounded 2\<rbrace>
+    refill_unblock_check sc_ptr
```
Comment (michaelmcinerney) [inline proof/invariant-abstract/DetSchedSchedule_AI.thy:4375]: Indentation

**Review by michaelmcinerney — APPROVED** [review]

File: proof/invariant-abstract/DetSchedSchedule_AI.thy (hunk 1 above)
Comment (michaelmcinerney) [reply proof/invariant-abstract/DetSchedSchedule_AI.thy:3822]: (This was probably moved, so I'm definitely not bothered if it's left as is)

**Comment by mktnk3** [comment]
Thanks for the approval. Yes, I think I will merge this as is for now, once the test finishes green.

**Pull request merged**